repo2docker/buildpacks/nix/__init__.py

```diff
@@ -0,0 +1,92 @@
+"""BuildPack for nixpkgs environments"""
+import os
+
+import jinja2
+
+from ..base import BaseImage
+
+
+TEMPLATE = r"""
+FROM alpine
+
+USER root
+
+# Enable HTTPS support in wget.
+RUN apk add --update openssl ca-certificates && \
+    rm -r /var/cache/apk/*
+
+ARG NIX_VERSION="2.1.1"
+ARG NIX_SHA256="ad10b4da69035a585fe89d7330037c4a5d867a372bb0e52a1542ab95aec67999"
+
+# Set up user
+ARG NB_USER
+ARG NB_UID
+ENV USER ${NB_USER}
+ENV HOME /home/${NB_USER}
+
+RUN adduser --disabled-password \
+    --gecos "Default user" \
+    --uid 1000 \
+    ${NB_USER}
+WORKDIR /home/${NB_USER}
+
+EXPOSE 8888
+
+# Copy and chown stuff. This doubles the size of the repo, because
+# you can't actually copy as USER, only as root! Thanks, Docker!
```

Comment:
FYI this may be fixed by using https://docs.docker.com/develop/develop-images/multistage-build/#before-multi-stage-builds

Reply:
👍  Would be interesting to explore. Do you want to create a PR to get going on this? Otherwise we should create a new issue to track this.

Reply:
#273 would also be relevant for this.

Reply:
This is fixed in docker itself. We just need to add a check for the docker version, or bump our base supported docker version to 17.09 and use it: https://github.com/jupyter/repo2docker/issues/164